apps/src/lib/util/GoogleAnalyticsReporter.js

```diff
@@ -0,0 +1,156 @@
+window.dataLayer = window.dataLayer || [];
+
+/**
+ * Global function used to track Google Universal events.
+ * It is a wrapper class that appends information to window.dataLayer
+ * This is the equivalent of ga() in Google Universal
+ */
+function gtag() {
+  window.dataLayer.push(arguments);
+}
+
+/**
+ * Class used to control the initialization of Google analytics
+ * for code.org properties.
+ * The class is responsible for
+ * - Adding the Google Tags needed by Tag manager
+ * - Send any dimensions information
+ * - Wrap Google Universal and Analytics functions for easier transition
+ * - Centralize common code
+ */
+class GoogleAnalyticsReporter {
+  isGoogleAnalyticsUniversalEnabled = true;
+  isGoogleAnalytics4Enabled = true;
+  googleUniversalGTag = 'UA-37745279-1';
+  googleAnalyticsGTag = 'G-L9HT5MZ3HD';
+
+  constructor() {
+    this.#initializeGoogleUniversal(
+      window,
+      window.document,
+      'script',
+      '//www.google-analytics.com/analytics.js',
+      'ga'
+    );
+
+    this.#initializeGoogleAnalytics();
+
+    this.#addEventListener();
+
+    window.trackEvent = this.trackEvent;
+    window.readCookie = this.readCookie;
```

Comment:
I haven't found where this is used besides extracting the language for hour of code which I now do using the request -_- I think it is unnecessary code but gotta check.

I am very againts removing big functions with already big changes

Reply:
Were you able to learn any more about this?